test_runner/logic_test_util.py

```diff
@@ -67,9 +71,30 @@ def RunLogicTestOnSim(sim_id,
   developer_dir = os.environ.get('DEVELOPER_DIR')
   if developer_dir:
     simctl_env_vars['DEVELOPER_DIR'] = developer_dir
+
+  xctest_tool = xcode_info_util.GetXctestToolPath(ios_constants.SDK.IPHONESIMULATOR)
+
+  test_bundle_name = os.path.splitext(os.path.basename(test_bundle_path))[0]
+  test_executable = os.path.join(test_bundle_path, test_bundle_name)
+  test_archs = bundle_util.GetFileArchTypes(test_executable)
+  
+  # if a logic bundle is built w/ x86 on Apple silicon, it won't be able to launch on the ARM64 sim; rework the xctest to fix this
+  if ios_constants.ARCH.X86_64 in test_archs and len(test_archs) == 1:
+    replacement_xctest_tool = os.path.join(tempfile.mkdtemp(), "xctest")
+    bundle_util.LeaveOnlyArchType(
+      xctest_tool,
+      replacement_xctest_tool,
+      ios_constants.ARCH.X86_64
+    )
+    xctest_tool = replacement_xctest_tool
+    platform_developer_dir = os.path.join(xcode_info_util.GetSdkPlatformPath(ios_constants.SDK.IPHONESIMULATOR), "Developer")
+    simctl_env_vars[_SIMCTL_ENV_VAR_PREFIX + "DYLD_FALLBACK_LIBRARY_PATH"] = "{0}/usr/lib".format(platform_developer_dir)
```

Comment:
```suggestion
    simctl_env_vars[_SIMCTL_ENV_VAR_PREFIX + "DYLD_LIBRARY_PATH"] = "{0}/usr/lib".format(platform_developer_dir)
```

this fixes Xcode 13.4 support, and still works for earlier versions. I'm not sure why yet but something about the semantics of this changed.